src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -71,6 +72,10 @@ const FrontendEngineInner = forwardRef<IFrontendEngineRef, IFrontendEngineProps>
 		}
 	}, [checkIsFormValid, onChange, watch]);
 
+	useDeepCompareEffectNoCheck(() => {
+		reset({ ...defaultValues, ...getValues() });
```

Comment:
apparently we have to use `reset` to update the field values when `defaultValues` is changed. unfortunately `reset` will clear all values entered by user hence the solution is to combine both entered values and default values. I am giving the higher precedence for the entered values as they are explicitly defined by the user.